tests/cosmosLightClient/test-nim-light-client-in-cosmos.ts

```diff
@@ -0,0 +1,225 @@
+import { dirname } from 'node:path';
+import { fileURLToPath } from 'node:url';
+import { readFile } from 'fs/promises';
+import glob_ from 'glob';
+const glob = glob_.sync;
+import { promisify } from 'node:util';
+import { exec as exec_, execSync, spawn } from 'node:child_process';
+
+import { SigningCosmWasmClient } from '@cosmjs/cosmwasm-stargate';
+import { DirectSecp256k1HdWallet, OfflineSigner } from '@cosmjs/proto-signing';
+import { calculateFee, GasPrice } from '@cosmjs/stargate';
+import * as fs from 'fs';
+import { SSZSpecTypes } from '../../libs/typescript/ts-utils/sszSpecTypes.js';
+import { jsonToSerializedBase64 } from '../../libs/typescript/ts-utils/ssz-utils';
+import { compileNimFileToWasm } from '../../libs/typescript/ts-utils/compile-nim-to-wasm.js';
+
+const exec = promisify(exec_);
+
+describe('Light Client In Cosmos', () => {
+  const controller = new AbortController();
+  const { signal } = controller;
+
+  const rootDir = dirname(fileURLToPath(import.meta.url));
+
+  const rpcEndpoint = 'http://localhost:26657';
+  const gasPrice = GasPrice.fromString('0.0000025ustake');
+
+  let DendrETHWalletInfo = {
+    mnemonic:
+      'economy stock theory fatal elder harbor betray wasp final emotion task crumble siren bottom lizard educate guess current outdoor pair theory focus wife stone',
+    address: '',
+  };
+
+  let wallet: OfflineSigner, client: SigningCosmWasmClient;
+  let _contractAddress;
+  beforeAll(async () => {
+    let contractDir = rootDir + `/../../contracts/cosmos/light-client`;
+    let nimFilePath = contractDir + `/lib/nim/light_client_cosmos_wrapper.nim`;
+    await compileNimFileToWasm(
+      nimFilePath,
+      `--nimcache:${contractDir}/nimbuild --d:lightClientCosmos -o:${contractDir}/nimbuild/light_client.wasm`,
+    );
+
+    let compileContractCommand = `docker run -t --rm -v "${contractDir}":/code --mount type=volume,source="$(basename "$(pwd)")_cache",target=/code/target --mount type=volume,source=registry_cache,target=/usr/local/cargo/registry cosmwasm/rust-optimizer:0.12.8 .`;
+    console.info(`➤ ${compileContractCommand}`);
+    await exec(compileContractCommand);
+
+    const setupWasmdCommand = `bash ${rootDir}/../../contracts/cosmos/light-client/scripts/setup_wasmd.sh`;
```

Comment:
Here and elsewhere in the file, it would be good to place quotes around the path here because the `rootDir` variable may contain spaces.